pkg/cloud/api/mutable_resource.go

```diff
@@ -54,6 +54,13 @@ func (e *ConversionError) Error() string {
 	return fmt.Sprintf("ConversionError: missing fields %v", e.MissingFields)
 }
 
+// MissingTypeError is returned when type traits has missing type.
+type MissingTypeError struct{}
+
+func (m *MissingTypeError) Error() string {
```

Comment:
It would be better to be able to put more details in the error message to make it easy for someone who hits this error to know immediately where the error is coming from.

Are we going to be checking specifically for MissingTypeError? If not, can we just use fmt.Errorf() and give the user more meaningful errors than just MissingTypeError?